src/main/java/de/tum/in/www1/artemis/web/rest/lecture/VideoUnitResource.java

```diff
@@ -0,0 +1,139 @@
+     * @param videoUnit the video unit to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated videoUnit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("/lectures/{lectureId}/video-units")
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR')")
+    public ResponseEntity<VideoUnit> updateVideoUnit(@PathVariable Long lectureId, @RequestBody VideoUnit videoUnit) throws URISyntaxException {
+        log.debug("REST request to update an video unit : {}", videoUnit);
+        if (videoUnit.getId() == null) {
+            return badRequest();
+        }
+
+        if (videoUnit.getLecture() == null || videoUnit.getLecture().getCourse() == null) {
+            return conflict();
+        }
+
+        if (!authorizationCheckService.isAtLeastInstructorInCourse(videoUnit.getLecture().getCourse(), null)) {
+            return forbidden();
+        }
+
+        if (!videoUnit.getLecture().getId().equals(lectureId)) {
+            return conflict();
+        }
+
+        VideoUnit result = videoUnitRepository.save(videoUnit);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, videoUnit.getId().toString())).body(result);
+    }
+
+    /**
+     * POST /lectures/:lectureId/video-units : creates a new video unit.
+     *
+     * @param lectureId      the id of the lecture to which the video unit should be added
+     * @param videoUnit the video unit that should be created
+     * @return the ResponseEntity with status 201 (Created) and with body the new video unit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/lectures/{lectureId}/video-units")
+    @PreAuthorize("hasAnyRole('ADMIN','INSTRUCTOR')")
+    public ResponseEntity<VideoUnit> createVideoUnit(@PathVariable Long lectureId, @RequestBody VideoUnit videoUnit) throws URISyntaxException {
+        log.debug("REST request to create VideoUnit : {}", videoUnit);
+        if (videoUnit.getId() != null) {
+            return badRequest();
+        }
+        Optional<Lecture> lectureOptional = lectureRepository.findByIdWithStudentQuestionsAndLectureUnits(lectureId);
+        if (lectureOptional.isEmpty()) {
+            return badRequest();
+        }
+        Lecture lecture = lectureOptional.get();
+        if (lecture.getCourse() == null) {
+            return conflict();
+        }
+        if (!authorizationCheckService.isAtLeastInstructorInCourse(lecture.getCourse(), null)) {
+            return forbidden();
+        }
+
+        lecture.addLectureUnit(videoUnit);
+        Lecture updatedLecture = lectureRepository.save(lecture);
+        VideoUnit persistedVideoUnit = (VideoUnit) updatedLecture.getLectureUnits().get(updatedLecture.getLectureUnits().size() - 1);
+
+        return ResponseEntity.created(new URI("/api/video-units/" + persistedVideoUnit.getId()))
```

Comment:
an env-variable containing "/api/video-units" would be nice

Reply:
What do you mean by this? Some of the APIs start with /video-units but some with /lectures, so I can not add it to @RequestMapping. Do you have an example?